### ISSUE TITLE
Fix the usage of aria-haspopup 

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24796]
**Describe the bug**

The ARIA attribute `aria-haspopup` is used inconsistently across the Gutenberg UI. For example, some buttons that open a menu do use `aria-haspopup`, while others don't.

It is worth reminding that `aria-haspopup` changed between the ARIA 1.0 and 1.1 specifications.

In ARIA 1.0: https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-haspopup
> Indicates that the element has a popup context menu or sub-level menu.
> This means that activation (note by me: this means **activation by the user**) renders conditional content. Note that ordinary tooltips are not considered popups in this context.
> A popup is generally presented visually as a group of items that appears to be on top of the main page content.

In ARIA 1.1 it has been expanded to indicate opening of other patterns that aren't just menus: https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup

Where the allowed values are:
```
false (default)	Indicates the element does not have a popup.
true            Indicates the popup is a menu.
menu            Indicates the popup is a menu.
listbox         Indicates the popup is a listbox.
tree            Indicates the popup is a tree.
grid            Indicates the popup is a grid.
dialog          Indicates the popup is a dialog.
```

It is clear that the concept of "popup" evolved to represent other kind of UIs.

Regardless, right now in Gutenberg there's the need to consistently communicate to assistive technology users whether a UI control opens some kind of additional UI. Worth noting that `aria-expanded` alone isn't sufficient because it represents a _state_, while `aria-haspopup` is a _property_. 

Examples:
In the block toolbars, some buttons that open a menu do use `aria-haspopup`, for example:
- the block switcher button
- the text alignment button

Other buttons that open a menu don't use `aria-haspopup`, for example:
- the "Replace" button in the Image block

Screenshots:

<img width="865" alt="Screenshot 2020-08-25 at 12 47 54" src="https://user-images.githubusercontent.com/1682452/91188880-165eb100-e6f2-11ea-85c6-40e5880ffb77.png">

<img width="832" alt="Screenshot 2020-08-25 at 12 48 08" src="https://user-images.githubusercontent.com/1682452/91188894-19f23800-e6f2-11ea-92a1-001889441805.png">

<img width="837" alt="Screenshot 2020-08-25 at 12 48 24" src="https://user-images.githubusercontent.com/1682452/91188897-1a8ace80-e6f2-11ea-9b0e-f6b2fe35e75d.png">

There's also the case of a button that uses `aria-haspopup` but it opens... another toolbar. Semantically, this isn't correct and it can be confusing for users. I'll split this in a separate ticket. See the Heading block, where the "Change heading level" button opens another toolbar:

<img width="682" alt="aria-haspopup for a toolbar" src="https://user-images.githubusercontent.com/1682452/91189207-781f1b00-e6f2-11ea-8309-509465e1a8ca.png">

<img width="784" alt="Screenshot 2020-08-25 at 16 52 04" src="https://user-images.githubusercontent.com/1682452/91189990-5e320800-e6f3-11ea-9f79-015232d0c146.png">

Lastly, there are cases where a button opens a complex UI that is made of various parts. I'm not sure which of the `aria-haspopup` values would be the most appropriate in these cases. This would need some research and discussion. For example, see the case of the "Quick inserter" or the “Content structure” UIs. How can we properly communicate what kind of UI is going to be opened? At a first glance, the closest value might be "dialog" but this UIs aren't real ARIA dialogs.

<img width="416" alt="Screenshot 2020-08-25 at 15 58 18" src="https://user-images.githubusercontent.com/1682452/91190552-ff20c300-e6f3-11ea-8760-8310e6772e14.png">

**Comment by tellyworth** [comment]
> At a first glance, the closest value might be "dialog" but this UIs aren't real ARIA dialogs.

If we were to pick a value to start with that's better than nothing, would it be `dialog`? Or simply `true`?

**Comment by afercia** [comment]
When the opened UI isn't one of the cases listed by the ARIA 1.1 spec (menu, listbox, tree, grid, dialog) I'd say the safest option is `true`.

**Comment by jorgefilipecosta** [comment]
Hi @afercia, @tellyworth,
I conducted some tests within the editor. The styles I added using the developer tools proved useful:
```
[aria-expanded] {
    border: 2px solid red !important;
}
```

It appears that the link format, inserter, and list view were the remaining cases lacking the aria-haspopup and aria-expanded attributes. I've submitted PR's to rectify these concerns. Are there any other cases that require attention?

**Comment by jorgefilipecosta** [comment]
I'm closing the issue as it appears the remaining cases were addressed if there is any remaining issue feel free to reopen.

**Comment by afercia** [comment]
@jorgefilipecosta thanks much for working on this. Also, nice to see you again.
Sure if we will notice remaining issues with these attributes we will either reopen or report separately.
